libvast/src/table_slice.cpp

```diff
@@ -54,6 +61,47 @@ record_type table_slice::layout(size_type first_column,
   return record_type{std::move(sub_records)};
 }
 
+table_slice_ptr table_slice::make(record_type layout, caf::actor_system&,
+                                  caf::atom_value impl) {
+  // TODO: extend CAF to allow storing arbitrary factory functions and remove
+  //       hardwired 'DEFAULT' code here.
+  if (impl == caf::atom("DEFAULT"))
+    return caf::make_counted<default_table_slice>(std::move(layout));
+  return nullptr;
+}
+
+caf::error table_slice::serialize_ptr(caf::serializer& sink,
+                                      const_table_slice_ptr ptr) {
+  if (!ptr) {
+    record_type dummy;
+    return sink(dummy);
+  }
+  return caf::error::eval([&] { return sink(ptr->layout()); },
+                          [&] { return sink(ptr->implementation_id()); },
+                          [&] { return ptr->serialize(sink); });
+}
+
+caf::error table_slice::deserialize_ptr(caf::deserializer& source,
+                                        table_slice_ptr& ptr) {
+  if (source.context() == nullptr)
+    return caf::sec::no_context;
+  record_type layout;
+  caf::atom_value impl_id;
+  auto err = caf::error::eval([&] { return source(layout); },
+                              [&] { return source(impl_id); });
+  if (err)
+    return err;
+  // Only default-constructed table slice handles have an empty layout.
+  if (layout.fields.empty()) {
+    ptr.reset();
+    return caf::none;
+  }
+  ptr = make(std::move(layout), source.context()->system(), impl_id);
```

Comment:
It seem that `make` does not make use of the `actor_system` at the moment. Will it (probably the contained `actor_system_config`) eventually contain a concrete factory?

Reply:
Yes, eventually we'll get a factory function from the system. By having the argument now we won't need to refactor calling code later.